front/lib/plans/subscription.ts

```diff
@@ -0,0 +1,247 @@
+import { Op } from "sequelize";
+
+import { front_sequelize, ModelId } from "@app/lib/databases";
+import { Plan, Subscription } from "@app/lib/models";
+import {
+  FREE_TEST_PLAN_DATA,
+  FREE_UPGRADED_PLAN_CODE,
+  PlanAttributes,
+} from "@app/lib/plans/free_plans";
+import { generateModelSId } from "@app/lib/utils";
+import logger from "@app/logger/logger";
+import { SubscribedPlanType } from "@app/types/user";
+
+export const getActiveWorkspacePlan = async ({
+  workspaceModelId,
+}: {
+  workspaceModelId: ModelId | null;
+}): Promise<SubscribedPlanType> => {
+  let activeSubscription: Subscription | null = null;
+  const today = new Date();
+  today.setHours(0, 0, 0, 0);
+
+  if (workspaceModelId) {
+    activeSubscription = await Subscription.findOne({
+      where: {
+        workspaceId: workspaceModelId,
+        startDate: { [Op.lte]: today },
```

Comment:
Let's not use date to define what is the current workspace.

The start and end date are logging information and should not be used to determine the current subscription.

We want a status. There is one active subscription and many other with a different status and here we search for the "active" one.